File: leetsync/1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.go
func check(nums []int) bool {
	n := len(nums)
	k := 0
	for i := 0; i < n; i++ {
		if nums[i] > nums[(i+1)%n] {
			k++
		}
		if k > 1 {
			return false
		}
	}
	return true
}

File: leetsync/1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;
        int k = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] > nums[(i + 1) % n]) {
                k++;
            }
            if (k > 1) {
                return false;
            }
        }
        return true;
    }
}